### PR TITLE
added method download mod (tarball) from github

### PR DESCRIPTION
**Pull request by BRO3886**
* generates a 'modules' folder
* inside the modules folder, it separates by source, such as "api.github.com"
* inside the source folder there's separation for user/repo

* downloads the tarball
* logs an error if unable to download

* added access token example as well (sample.env)

### CODE REVIEW

File: .env.sample [hunk 1]
```diff
@@ -0,0 +1 @@
+GITHUB_ACCESS_TOKEN=<YOUR ACCESS TOKEN>
```
Comment (SuperFola) [inline .env.sample:1]: I went with this kind of strategy (I forgot to push it long ago, and wondered where it was when I reviewed your PR) https://github.com/ArkScript-lang/nuclear/blob/master/nuclear/__main__.py#L67-L84, maybe it's not the way to go and .env files are better?

File: nuclear/install.py [hunk 2]
```diff
@@ -31,6 +36,13 @@ def handle(args: argparse.Namespace) -> int:
...
     if tar_addr is not None:
         # download
-        pass
-
+        r = requests.get(tar_addr)
+        # get filename from content-dispoisiton
+        filename = get_dir.get_filename(r.headers.get('content-disposition'))
+        try:  
+            makedirs(get_dir.get_dir_name(tar_addr))
+            open(get_dir.get_dir_name(tar_addr)+"/"+filename,'xb').write(r.content)
```
Comment (SuperFola) [inline nuclear/install.py]: maybe do it using `with` so that the opened file get closed at the end

**Review by SuperFola — CHANGES_REQUESTED** [review]

**Comment by SuperFola** [comment]
About that part:

> * generates a 'modules' folder
> * inside the modules folder, it separates by source, such as "api.github.com"

The package manager shouldn't work like node, nor like pip, but install the packages in a central "local repository" (somewhere like `~/ark-pkgs` for example, which could be left to be configured by the end user, referenced in a sort of packages.json per project).

The idea to separate per source and per user is great, I would also add that the package manager was designed to be able to handle multiple version of the same package, so it would need an organization like `source/user/repo/version`.

Later, each installed module will be checked for any "packages.json" file (or whatever we'll call it) to install its dependencies (which shall get listed as well under the "packages.json" of the current project), so that the module loader can read this file and get the exact wanted file.

File: .env.sample (hunk 1 above)
Comment (BRO3886) [reply .env.sample:1]: this is a sample .env for people to clone and get started with, I am using a .env file only

File: nuclear/install.py (hunk 2 above)
Comment (BRO3886) [reply nuclear/install.py]: Okay I'll do this

File: .env.sample (hunk 1 above)
Comment (SuperFola) [reply .env.sample:1]: no problem, let's ditch the nuclear config file then

**Comment by BRO3886** [comment]
> About that part:
> 
> > * generates a 'modules' folder
> > * inside the modules folder, it separates by source, such as "api.github.com"
> 
> The package manager shouldn't work like node, nor like pip, but install the packages in a central "local repository" (somewhere like `~/ark-pkgs` for example, which could be left to be configured by the end user, referenced in a sort of packages.json per project).
> 
> The idea to separate per source and per user is great, I would also add that the package manager was designed to be able to handle multiple version of the same package, so it would need an organization like `source/user/repo/version`.
> 
> Later, each installed module will be checked for any "packages.json" file (or whatever we'll call it) to install its dependencies (which shall get listed as well under the "packages.json" of the current project), so that the module loader can read this file and get the exact wanted file.


Can you send some package links for me to work with? It'll help with the version-folder creation

File: README.md [hunk 3]
```diff
@@ -29,6 +29,22 @@ An ArkScript Package Manager
...
+```bash
+touch .env
+nano .env
+```
+* insert access token in the .env file
+```bash
+GITHUB_ACCESS_TOKEN=<YOUR ACCESS TOKEN>
+```
+* save and exit
+* Checkout [sample.env](./sample.env) to see how the .env file should look like
```
Comment (SuperFola) [inline README.md]: .env.sample

**Comment by SuperFola** [comment]
Also, could you get up to date with master?

----

As for the tests, alas we don't have any repositories, I'm just testing with ArkScript-lang/std (I will add a 0.0.1 release since we finally have documentation for all of the functions).

**Comment by BRO3886** [comment]
Solves #1 and #3 I guess

**Review by SuperFola — COMMENTED** [review]
Seems good to me, I'll test that tomorrow and it should be merged!

Thanks for your contribution!

**Comment by BRO3886** [comment]
https://hacktoberfest.digitalocean.com/hacktoberfest-update?updated

Can you add the appropriate tags please?

**Comment by SuperFola** [comment]
Hey! AFAIK, no need to add a topic to the PR since it's merged and the repo is correctly labelled!

**Pull request merged**